### PR TITLE
docs: document that systemd closes console during normal runtime

### LINKED ISSUE #21451: systemd closes container /dev/console
**systemd version the issue has been seen with**
 > 249.6

**Linux kernel version used** (`uname -a`)
 > Linux d 5.15.1-1-ARCH #1 SMP Sun Nov 7 14:40:17 MST 2021 aarch64 GNU/Linux

**CPU architecture issue was seen on**
 > aarch64

**Unexpected behaviour you saw**

systemd boots fine within a container (unshare), but then after "[  OK  ] Reached target Graphical Interface." it closes /dev/console which closes the controling pty. 

    panic: read /dev/ptmx: input/output error

i'm unsure if this is intended behaviour, if yes it would be great if it was documented in
https://www.freedesktop.org/wiki/Software/systemd/ContainerInterface/
specifically what kind of tty is expected in /dev/console. 
binding the host /dev/console seems to work fine, but that's of course not the intended use case with containers.

also just ignoring the EOF works, but it's unclear if this is intended

**Steps to reproduce the problem**

```
#!/bin/sh
set -ex

mkdir -p /tmp/sysd
mount none -t tmpfs /tmp/sysd
mkdir -p /tmp/sysd/root
mount --bind /  /tmp/sysd/root/ -o ro
mkdir -p /tmp/sysd/root/proc
mount none -t tmpfs /tmp/sysd/root/dev/

for i in null zero full random urandom tty ptmx
do
    touch /tmp/sysd/root/dev/$i
    mount --bind /dev/$i /tmp/sysd/root/dev/$i
done

touch /tmp/sysd/root/dev/console
mount --bind $(tty)  /tmp/sysd/root/dev/console
mount --bind /sys     /tmp/sysd/root/sys -o ro
mount --bind /proc   /tmp/sysd/root/proc -o ro

container=docker unshare --mount-proc -pf -R /tmp/sysd/root /usr/lib/systemd/systemd
```

### PR DESCRIPTION
**Pull request by aep**
fixes #21451

Signed-off-by: Arvid E. Picciani <aep@exys.org>

### CODE REVIEW

File: docs/CONTAINER_INTERFACE.md [hunk 1]
```diff
@@ -37,7 +37,10 @@ manager, please consider supporting the following interfaces.
    in this context.)
 
 3. Pre-mount `/dev/` as (container private) `tmpfs` for the container and bind
-   mount some suitable TTY to `/dev/console`. Also, make sure to create device
+   mount some suitable TTY to `/dev/console`. If this is a pty, make sure to not
+   close the controling pty master during systemd lifetime. PID1 will close ttys,
+   to avoid being killed by SAK. it only opens ttys for the time it
```
Comment (keszybz) [inline docs/CONTAINER_INTERFACE.md]: Capitalize "it" here.

Comment (keszybz) [reply docs/CONTAINER_INTERFACE.md]: "systemd lifetime" → "systemd's lifetime"

**Review by keszybz — COMMENTED** [review]
Looks OK, except for the spelling issues.

File: docs/CONTAINER_INTERFACE.md [hunk 2]
```diff
@@ -37,7 +37,10 @@ manager, please consider supporting the following interfaces.
    in this context.)
 
 3. Pre-mount `/dev/` as (container private) `tmpfs` for the container and bind
-   mount some suitable TTY to `/dev/console`. Also, make sure to create device
+   mount some suitable TTY to `/dev/console`. If this is a pty, make sure to not
+   close the controling pty master during systemd's lifetime. PID1 will close ttys,
```
Comment (bluca) [inline docs/CONTAINER_INTERFACE.md:41]: Typo: controlling

**Pull request merged**